runner/client/src/mill/runner/client/MillProcessLauncher.java

```diff
@@ -248,15 +249,31 @@ static int getTerminalDim(String s, boolean inheritError) throws Exception {
 
   private static AtomicReference<String> memoizedTerminalDims = new AtomicReference();
 
+  private static final boolean canUseNativeTerminal;
+
+  static {
+    boolean canUse;
+    if (mill.main.client.Util.hasConsole())
+      try {
+        NativeTerminal.getSize();
+        canUse = true;
+      } catch (Throwable ex) {
+        canUse = false;
+      }
+    else canUse = false;
+
+    canUseNativeTerminal = canUse;
+  }
+
   static void writeTerminalDims(boolean tputExists, Path serverDir) throws Exception {
     String str;
 
     try {
-      if (java.lang.System.console() == null) str = "0 0";
+      if (!mill.main.client.Util.hasConsole()) str = "0 0";
```

Comment:
It seems we're checking `mill.main.client.Util.hasConsole()` twice here; once in the `static` blockl and once in `writeTerminalDims`. We can probably skip one of those checks right?

Reply:
`hasConsole` caches its result now, so calling it twice shouldn't incur any cost